packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -274,6 +275,13 @@ class TransactionService {
             tr.spans.push(createTotalBlockingTimeSpan(metrics.tbt))
           }
         }
+        /**
+         * Categorize the transaction based on the current location
+         */
+        if (tr.name === NAME_UNKNOWN) {
+          tr.name = slugifyUrl(window.location.href)
+        }
```

Comment:
There might be a race condition here, since we execute this in the promise, there's a possibility that we capture the url for the next route if the route change happens before this point.

Reply:
But that would be redefined and we would still end up capturing the correct paths provided by the `route-change` transactions. So i guess we are safe here.

Reply:
Not sure if that helps, since the href will be changed by the routing framework.

Reply:
For all the instrumented frameworks, this does not create any issues as you see from the E2E tests. However for non-supported frameworks or routers, we will end up capturing `href` and its anyways a better solution than having `UNKNOWN`. Do you agree?